frankenphp.go

```diff
@@ -359,6 +362,13 @@ func go_fetch_request() C.uintptr_t {
 	return C.uintptr_t(cgo.NewHandle(r))
 }
 
+func maybeCloseContext(fc *FrankenPHPContext) {
+	if !fc.closed {
+		close(fc.done)
+		fc.closed = true
+	}
+}
+
```

Comment:
It appears we close this channel in a couple of places in the code, which isn't ideal when we want to make sure we close it exactly once.

Reply:
You can probably use https://pkg.go.dev/sync#Once for this use case.